### DISCUSSION TITLE
demo does not run

### DISCUSSION THREAD

**Discussion by SEXTEDDY** [CodinGame/monaco-vscode-api#225] [General]
cd demo 
npm ci
npm start

![image](https://github.com/CodinGame/monaco-vscode-api/assets/29817807/b22f3f4b-d4e9-467c-a2d3-57bea4fb7db8)

**Comment by CGNonofr** [comment]
For the demo to work, the main project needs to be built (`npm ci && npm run build` at the root)

I'll add that "detail" in the README

**Reply by SEXTEDDY** [reply]
Thank you for your response, I am not a professional nodejs developer and may ask some very simple questions.

When I execute "npm ci" in the root directory, I get the following error
![image](https://github.com/CodinGame/monaco-vscode-api/assets/29817807/cc26f411-2f70-42b3-919f-8d861502d64d)

**Reply by kaisalmen** [reply]
@SEXTEDDY You can't build this project on Windows directly, but it works with the [Windows Subsystem for Linux](https://learn.microsoft.com/en-us/windows/wsl/about)

**Reply by CGNonofr** [reply]
Ok I'm not sure you'll be able to run it on a Window. I've never tested it

Maybe try inside a container

Otherwise, you can replace all demo local dependencies but their published versions:

`"@codingame/monaco-vscode-accessibility-service-override": "file:../dist/service-override-accessibility"` becomes `"@codingame/monaco-vscode-accessibility-service-override": "1.83.5"` (and the same for all `file:../` dependencies) (`"vscode": "file:../dist/main"` becomes `"vscode": "npm:@codingame/monaco-vscode-api@1.83.5"`) then `npm install && npm start`

**Reply by SEXTEDDY** [reply]
Still error. Should I install npm under wsl. 
![image](https://github.com/CodinGame/monaco-vscode-api/assets/29817807/fa8b2772-0586-4c36-8239-49df212466da)

**Reply by kaisalmen** [reply]
Is the script executable? Try `chmod 755 ./scripts/install-vscode`

**Reply by SEXTEDDY** [reply]
Without any information, I suppose it should be executable?
![image](https://github.com/CodinGame/monaco-vscode-api/assets/29817807/ea011a3c-61c1-4c40-9d4a-552a0ba6be56)

**Reply by SEXTEDDY** [reply]
I changed "file:" to" ~1.83.5 "but executing npm install failed
![image](https://github.com/CodinGame/monaco-vscode-api/assets/29817807/fc8aa736-cacd-46df-8853-791e6f0cc855)

**Reply by CGNonofr** [reply]
> Api rate limit exceeded

**Reply by SEXTEDDY** [reply]
Do you use macos or linux? I'll get a computer and try it out

**Reply by SEXTEDDY** [reply]
I am very sorry to bother you, but I have another question to consult. The following error occurs when I execute npm start after successfully running npm install &npm run build
![image](https://github.com/CodinGame/monaco-vscode-api/assets/29817807/9500100b-1885-4eda-9b78-1a191c69c85c)


I feel it has something to do with the vite you talked about in the readme, but I really don't know how to deal with it
![image](https://github.com/CodinGame/monaco-vscode-api/assets/29817807/c3bd595e-29b4-4c2c-acc6-471ee678781f)


Look forward to your help! thank you

**Reply by CGNonofr** [reply]
I'm unable to reproduce, what is your env? and what make you think it's related to the import.meta.url plugin?
